### PR TITLE
Feature/views

### PR DESCRIPTION
**Pull request by ShinhyoBelok**
# Changes 🔨 

* Add root path in routes config `root to: "users#index"`.
* Add root view template.
* Add view template to `"/users/:id"`
* Add view template to `"/users/:user_id/posts"`
* Add view template to `"/users/:user_id/posts/:id"`

## Add seed test database.

To create a test database, you can run `rails db:seed` and will create the following data:

image

### CODE REVIEW

File: app/views/users/_user_card.html.erb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<article class="user_card">
+    <img src="https://pm1.narvii.com/7258/5520799cf0539b408bd8abee0a14d3a492ee5107r1-753-753v2_hq.jpg" alt="User photo">
+    <div class="user_info">
+        <h2><%= user.name %></h2>
+        <h4>Number of post: <%= user.posts_counter %></h4>
+    </div>
+</article>
```
Comment (BertrandConxy) [inline app/views/users/_user_card.html.erb]: - [x] Good job so far 👏 It looks like you are trying to render a static image for each user while every user should have a unique image. Please add a user link in the image tag and it should probably look like this ` alt="User photo">`

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1 +1,6 @@
-<h1> List of user name and the number of posts <h1>
\ No newline at end of file
+<%= stylesheet_link_tag('user_card') %>
+<section>
+    <% @users.each do |user| %>
+    <%= render "user_card", user: user %>
+    <%end%>
+</section>
```
Comment (BertrandConxy) [inline app/views/users/index.html.erb]: - [ ] Currently, on the home page there is nothing being displayed there due to the error that I didn't notice. This is how it looks 👇 
image

I have added some seed data but I can see them. Due to that, I can't proceed to check the posts and comments, so please correct it 🙏

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1 +1,12 @@
...
+<section class="user_detail">
+    <%= render "user_card", user: @user %>
+    <article class="bio border">
+        <h2>Bio</h2>
+        <p><%= @user.bio %></p>
+    </article>
+    <% @user.posts.each do |post| %>
+    <%= render "posts/post_card", post: post %>
+    <%end%>
+    <button class="see_all">See all posts</button>
```
Comment (BertrandConxy) [inline app/views/users/show.html.erb]: - [x] From the project requirements, you are expected to run the custom method `recent_posts`  to render 3 recent posts so, please change that 😃 👍

File: app/views/posts/_post_card.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<article class="post border">
+    <h3 class="title"><%= post.title %></h3>
+    <p><%= post.text %></p>
+    <div class="counter">
+        <h5>Comments: <%= post.comments_counter %></h5>
+        <h5>Likes: <%= post.likes_counter %></h5>
+    </div>
+</article>
```
Comment (BertrandConxy) [inline app/views/posts/_post_card.html.erb]: - [x] From the project requirements, clicking the post will redirect the user to the post, however, I can see that there is no link there, please add that in order to fulfill the project requirements 😃

File: app/views/users/show.html.erb (hunk 3 above)
Comment (BertrandConxy) [inline app/views/users/show.html.erb]: - [x] the `See all posts` button is expected to redirect the user to all the user posts, therefore, I would suggest you add the link to that button. As a HINT 👇 

```ruby
<button>
<%= link_to 'See all Post', user_posts_path(@user) %>
</button>
```

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -1 +1,19 @@
...
+        </div>
+    </article>
+    <% @user.posts.each do |post| %>
+    <%= render "post_card", post: post %>
+    <% post.comments.each do |comment| %>
+    <ul class="comments_list">
+        <%= render "comment_card", comment: comment %>
+    </ul>
+    <%end%>
+    <%end%>
```
Comment (BertrandConxy) [inline app/views/posts/index.html.erb]: - [x] From the project requirements, here you are expected to render the 5 recent comments for each post using the custom method built in the previous milestone, `recent_comments` Please use it in order to fulfill the project requirements 👍

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @ShinhyoBelok, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 


## Highlights 💪 
- No linter errors
- correct flow is used


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @ShinhyoBelok 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**